tests/python/test_basic.py

```diff
@@ -271,14 +272,14 @@ def test_DMatrix_init_from_path(self):
 
     def test_DMatrix_save_to_path(self):
         """Saving to a binary file using pathlib from a DMatrix."""
-        data = np.random.randn(100, 2)
-        target = np.array([0, 1] * 50)
-        features = ['Feature1', 'Feature2']
+        data: np.ndarray = np.random.randn(100, 2)
+        target: np.ndarray = np.array([0, 1] * 50)
+        features: List[str] = ['Feature1', 'Feature2']
 
-        dm = xgb.DMatrix(data, label=target, feature_names=features)
+        dm: xgb.DMatrix = xgb.DMatrix(data, label=target, feature_names=features)
 
         # save, assert exists, remove file
-        binary_path = Path("dtrain.bin")
+        binary_path: Path = Path("dtrain.bin")
```

Comment:
I'm happy with these changes. But for the ease of working on type hints and consistency with the rest of the library, trivially deducible types don't need annotation.